### ISSUE TITLE
Error in the documentation regarding signals

### ISSUE THREAD

**Issue by E-Mello** [godotengine/godot-docs#9119]
**Your Godot version: 3.5.3 and 4.2.1**

**Issue description:**
I was following the documentation and noticed an error, I tried to perform the same commands that the documentation guides, both in versions 3.5.3 and 4.2.1 and both didn't work, when searching, I managed to do it another way.

Code provide by documentation:
```gdscript
extends Sprite2D

var speed = 400
var angular_speed = PI


func _ready():
	var timer = get_node("Timer")
	timer.timeout.connect(_on_timer_timeout)


func _process(delta):
	rotation += angular_speed * delta
	var velocity = Vector2.UP.rotated(rotation) * speed
	position += velocity * delta


func _on_button_pressed():
	set_process(not is_processing())


func _on_timer_timeout():
	visible = not visible
```
Code that worked:
```gdscript
extends Sprite

var speed = 400
var angular_speed = PI


func _ready():
	var timer = get_node("Timer")
	if timer:
		timer.connect("timeout", self, "_on_timer_timeout")

func _process(delta):
	rotation += angular_speed * delta
	var velocity = Vector2.UP.rotated(rotation) * speed
	position += velocity * delta


func _on_button_pressed():
	set_process(not is_processing())


func _on_timer_timeout():
	visible = not visible
```

I'm starting now.. so, I could be wrong, but the way the documentation describes the use of signal is apparently wrong and for those who are starting, it should delay a little

**URL to the documentation page (if already existing): https://github.com/godotengine/godot-docs/blob/4.2/getting_started/step_by_step/signals.rst**

**Comment by AThousandShips** [comment]
Since the issue seems to just be that part (the check for `timer`) and the confusion over that 4.x code doesn't work in 3.x, this is a duplicate of:
* https://github.com/godotengine/godot-docs/issues/8169

Which has been solved and will be cherry picked for 4.2 so it also gives the clarification that you should run the `Node2D` scene (the scene selected in the screenshot)

Thank you for your report, if I have missed something please say so and we can reopen this, but there doesn't seem to be any parts of this that aren't covered :)

**Comment by E-Mello** [comment]
wow, sorry, I must have got something wrong, I did it again in both versions and in fact you are absolutely right. Have a great day !
![Captura de tela de 2024-03-19 10-00-55](https://github.com/godotengine/godot-docs/assets/52728995/9e1367e2-c20b-4421-a41a-c20784654b80)
